### DISCUSSION TITLE
What is the cause of the delay restoring the window placement and their state

### DISCUSSION THREAD

**Discussion by W7STF** [kangyu-california/PersistentWindows#217] [Q&A]
Finally, a utility that I needed because I'm using two computers with two DP monitors and a KVM switch. I thought, surely Microsoft would have taken care of this. Well, evidently they have in Win11, but that is not what my company runs, nor myself on my personal computer, so that's not an option.

So I haven't looked at the source code yet, but what is the reason for the otherwise slow restore? Is this programmed to add this delay as default? Is it because of the code it leverages?

Seems like there might be some optimization possible.
Best regards.

**Comment by kangyu-california** [comment]
The slow restore might be due to default low priority of task. 
You may raise the priority of PW task to speed up restore, as discussed in #206
However some users have experienced incomplete restore due to adversarial move by Windows OS (#201, #212), especially if the monitor is slow to recover from sleep. In those cases, the workaround is to delay restore using -delay_auto_restore command option

**Comment by W7STF** [comment]
Kang, thanks for the response. So I can try upping the priority of the task.
One thing I though about experimenting with is pumping the keyboard event queue and hooking the messages related to keyboard connects and disconnects.

For example, without running a problem like your utility, I can have windows open in computer 1, used WinKey + M to minimize all windows, switch to computer 2, then when switching back to computer 1 (my KVM switches with a double-tap of the right CTRL key), I can then unminimize using WinKey + SHIFT + M and all the windows are restored to their place.

Works fine, but not for all windows. For example, programs like CPUID HWMonitor and those of HWINFO64 don't get restored to their proper locations. But for most all the other things I do and use, that's not a big drawback.

Can you recommend a sample WIN API example in C, C++ or C#, that I could perform the following:
keybd_event(VK_LWIN, 0, 0, 0);
keybd_event('M', 0, 0, 0);
keybd_event(VK_LWIN, 0, KEYEVENTF_KEYUP, 0);

and also pursue something like that found at:
https://stackoverflow.com/questions/9930958/win32-determining-when-keyboard-is-connected-disconnected
S.

**Reply by kangyu-california** [reply]
C# example to simulate keyboard event
https://stackoverflow.com/questions/14395377/how-to-simulate-a-ctrl-a-ctrl-c-using-keybd-event
